File: java/src/strategy/models/InsuranceCompany.java
package strategy.models;

import strategy.evaluators.InsuranceEvaluator;
import strategy.evaluators.InvalidVehicleEvaluator;


/**
 * InsuraceCompany
 *  Is the Context class
 *  Has a reference to a Strategy class
 *  Has an InvalidVehicleEvaluator Strategy by Default
 *  Is the used by the Client
 *  Passes Data (an integer) to the Strategy;
 * */
public class InsuranceCompany {

    private InsuranceEvaluator insuranceEvaluator;
    private int insuranceCompanyFee = 5;

    public InsuranceCompany(){
        insuranceEvaluator = new InvalidVehicleEvaluator();
    }

    public void setInsuranceEvaluator(InsuranceEvaluator evaluator){
        this.insuranceEvaluator = evaluator;
    }

    public int getQuoteForVehicle(){
        return insuranceEvaluator.getQuoteForVehicle(insuranceCompanyFee);
    }
}

File: js/src/strategy/InsuranceCompany.js
var InvalidInsuranceEvaluator = require("./evaluators/InvalidInsuranceEvaluator");

function InsuraceCompany(){
    var that = this;
    this.insuranceFee = 5;
    this.insuranceEvaluator = InvalidInsuranceEvaluator;
  return {    
      
      setInsuranceEvaluator : function(evaluator){
          that.insuranceEvaluator = evaluator;
          
      },
      quoteVehicle: function(){
           return that.insuranceEvaluator.getQuote(that.insuranceFee);
        }
   };
}

module.exports = InsuraceCompany; 